ultravox/training/config_base.py

```diff
@@ -80,6 +107,16 @@ class TrainConfig:
     loss_config: Optional[ultravox_config.LossConfig] = None
 
     def __post_init__(self):
+        if self.data_dicts:
+            self.data_dicts = [
+                DataDictConfig(**data_dict) for data_dict in self.data_dicts
+            ]
+            if self.data_sets:
+                self.data_sets.extend(self.data_dicts)
```

Comment:
nit and I guess we're not fixing it in this PR: I'm still not happy with this format. For example, you have to know that `data_dicts` get added on the right which is arbitrary and all the type hints are wrong.

Reply:
We will address this in the right way in a later PR and you will be happy :-)